caikit/runtime/service_factory.py

```diff
@@ -178,9 +177,18 @@ def _get_and_filter_modules(
         clean_modules = set()
         modules = [
             module_class
-            for module_class in caikit.core.registries.module_registry().values()
+            for module_class in registries.module_registry().values()
             if module_class.__module__.partition(".")[0] == lib
         ]
+        # NB: The `module_registry` only includes the `LOCAL` backend modules.
```

Comment:
I think this is the right implementation, but it's worth pointing out that we _could_ go down the rabbit hole of only merging in module implementations for backends that have been enabled at boot time. That said, I think that would be a gnarly bunch of config introspection given the way that module backends are managed by the `LOCAL` implementation of the `ModelInitializer` abstraction.

Reply:
I'm not sure we'd actually want to do that, since then the interfaces that are generated would depend on the runtime configuration. I had been assuming that the interfaces generated from a library should be consistent, modulo any explicit configurations in `runtime.service_generation`